envi/archs/i386/emu.py

```diff
@@ -112,11 +122,15 @@ def __init__(self, archmod=None):
 
         # Add our known calling conventions
         self.addCallingConvention('stdcall', stdcall)
-        self.addCallingConvention('thiscall', thiscall)
         self.addCallingConvention('cdecl', cdecl)
+        self.addCallingConvention('thiscall', thiscall)
         self.addCallingConvention('msfastcall', msfastcall)
         self.addCallingConvention('bfastcall', bfastcall)
 
+        self.addCallingConvention('thiscaller', thiscaller)
```

Comment:
renaming this could cause backward compatibility breaks, are they strictly necessary?

Reply:
Also, the name "thiscall" or "fastcall" are more common conventions that "*caller"

Reply:
So I'm not actually renaming thiscall/fastcall/etc to *caller. I'm adding a caller cleanup version of them since I keep running into caller cleanup versions of them where it's the calling function that cleans up the function parameters instead. Otherwise the emulator assumes it callee cleanup, cleans up the parameters, and then the calling function also cleans up the parameters, and that completely throws off where the stack pointer actually is, which induces a fun set of ripple effects for several of the analysis passes.

But yea, I wasn't super happy with the names (but I didn't want to conflict with the existing names). So I can swap the names on the new ones around.

Reply:
ThisCall_Special and thiscall_special, or "thiscall_ccu" or something else.  the "er" is confusing.  leave the originals as they are, and create the new ones with names that indicate that they're special.